client/src/infographic/TableOfContents.js

```diff
@@ -2,105 +2,68 @@ import _ from "lodash";
 import React from "react";
 
 import {
-  Details,
+  StatelessDetails,
   create_text_maker_component,
   UnlabeledTombstone,
+  LinkStyled,
 } from "src/components/index.js";
 
 import { separatorColor } from "src/core/color_defs.js";
 
-import { infograph_options_href_template } from "./infographic_link.js";
-
 import text from "./TableOfContents.yaml";
 
 const { TM } = create_text_maker_component(text);
 
 export default class TableOfContents extends React.Component {
-  state = {
-    initial_open: false,
-  };
-
-  handle_click = () => {
-    this.setState((prev_state) => ({
-      initial_open: !prev_state.initial_open,
-    }));
-  };
```

Comment:
I might've asked about this before but:

Since state change is asynchronous, wouldn't it be preferable to be using the `prev_state` instead of directly accesing `this.state`? Although, I guess it's next to impossible for this to fail considering that there's no other state change called in this component haha

Reply:
My mental model is as follows (it's been a while since I fact checked it though, I could be wrong, bullets starting with ?'s are things I'm going to go try and confirm after this):
  - `setState` calls are applied async, yes
  - React will, timing dependant, batch multiple `setState` calls together
  - the batched `setStates` only result in one change to the actual state (`this.state`)
  - ? but as each is applied they will create a new "pending" state (need to double check)
  - ? this "pending" state is what's actually passed as the `prev_state` arg to any callback `setState`s in the batch (need to double check)
  
The shakeout of that, assuming I'm correct, is a little tricky. It's going to be case dependant, and possibly not clear, on whether to use a callback with `prev_state` or directly access `this.state`. In this case though, I think any user interaction with a `Details` drawer that is rendered as open  (i.e. `this.state.is_open === true`) should ONLY be able to result in it closing and vice-versa. Updating the state based on the _actual_ state value forces that (even if multiple `setState`s end up in the same batch). Using the (incrementally updated/pending) `prev_state` value would cause the ultimate state to appear unchanged if an even number of calls end up in the same batch.

Make sense? I'll get back to you if it turns out those assumptions I listed were wrong though.

Reply:
Yea I think that makes sense 👍

Reply:
Everything seems to check out except my assumption that batching is mostly timing dependant. Haven't dug all the way down on how things specifically work in React 17, but all `setState` calls are batched and in at least some cases that batching is based on the scope they're kicked off from (e.g. all `setState` calls within the same react DOM event callback go in the same batch). If there's any exceptions/specific cases with dependence on timing, it's probably introduced by schedulers like `setTimeout`, `_.debounce`, etc.